akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala

```diff
@@ -110,3 +124,52 @@ object BehaviorInterceptor {
   }
 
 }
+
+/**
+ * A behavior interceptor allows for intercepting signals reception and perform arbitrary logic -
+ * transform, filter, send to a side channel etc.
+ *
+ * The `BehaviorSignalInterceptor` API is considered a low level tool for building other features and
+ * shouldn't be used for "normal" application logic. Several built-in intercepting behaviors
+ * are provided through factories in the respective `Behaviors`.
+ *
+ * If the interceptor does keep mutable state care must be taken to create a new instance from
+ * the factory function of `Behaviors.intercept` so that a new instance is created per spawned
+ * actor rather than shared among actor instance.
+ *
+ * @tparam Inner The inner message type - the type of message the wrapped behavior accepts
+ *
+ * @see [[BehaviorInterceptor]]
+ */
+abstract class BehaviorSignalInterceptor[Inner] extends BehaviorInterceptor[Inner, Inner](null) {
+  import BehaviorInterceptor._
+
+  /**
+   * Override to intercept actor startup. To trigger startup of
+   * the next behavior in the stack, call `target.start()`.
+   * @return The returned behavior will be the "started" behavior of the actor used to accept
+   *         the next message or signal.
+   */
+  override def aroundStart(ctx: TypedActorContext[Inner], target: PreStartTarget[Inner]): Behavior[Inner] =
```

Comment:
do we need to override this, looks the same impl

Reply:
thanks, good catch